Parsers/ASimWebSession/Parsers/ASIMWebSessionIIS.yaml

```diff
@@ -0,0 +1,110 @@
+Parser:
+  Title: Web Session ASIM parser for Windows IIS logs
+  Version: '0.1'
+  LastUpdated: Dec 13, 2022
+Product:
+  Name: Internet Information Services (IIS)
+Normalization:
+  Schema: WebSession
+  Version: '0.2.5'
+References:
+- Title: ASIM Web Session Schema
+  Link: https://aka.ms/ASimWebSessionDoc
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
+Description: |
+  This ASIM parser supports normalizing IIS logs produced to the ASIM Web Session normalized schema.
+ParserName: ASIMWebSessionIIS
+EquivalentBuiltInParser: _ASIM_WebSession_IIS
+ParserParams:
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let parser = (disabled: bool = false)
+        {
+        W3CIISLog
+        | extend
+                EventResult = iff ( toint(scStatus) < 400, "Success", "Failure"),
+                scStatus = toint(scStatus), 
+                csUriQuery = iff(csUriQuery == "-", "", csUriQuery),
+                csUserName = iff(csUserName == "-", "", csUserName),
+                HttpVersion = iff((csVersion has "HTTP"), split(csVersion, "/")[1], "") // there is a limited chance that something connects over non-HTTP
+        | project-rename 
+                HttpHost = sSiteName,
+                HttpRequestMethod = csMethod,
+                HttpReferrer = csReferer,
+                User = csUserName, //probably won't have this one often
+                Dvc = Computer,
+                EventResultDetails = scStatus,
+                Dst = sIP,
+                Src = cIP,
+                UserAgent = csUserAgent,
+                ThreatCategory = IndicatorThreatType,
+                EventOriginalSeverity = Severity,
+                SrcGeoCountry = RemoteIPCountry,
+                SrcGeoLatitude = RemoteIPLatitude,
+                SrcGeoLongitude = RemoteIPLongitude,
+                ThreatOriginalConfidence = Confidence,
+                ThreatIpAddr = MaliciousIP,
+                ThreatIsActive = IsActive,
+                ThreatFirstReportedTime = FirstReportedDateTime,
+                ThreatLastReportedTime = LastReportedDateTime,
+                EventReportUrl = ReportReferenceLink,
+                EventOriginalSubType = scSubStatus
+        | extend
+                DstIpAddr = iff(csHost <> "", split(csHost, ":")[0], Dst),
```

Comment:
csHost can be the host header and not just the IP address. Actually this is the preferred configuration. You would need to add a check if csHost is an IP address, and act accordingly.
```
| extend csHost_parts = split (csHost, ":")
| extend ip_part = tostring(csHost_parts[0])
| extend isIpAddr = ipv4_is_match(ip_part, "0.0.0.0",0) or ipv6_is_match(ip_part, "::1",0)
```

Reply:
This will not work for ipv6, do we have some examples how we tackled this in other parsers?

Reply:
It does work for IPv6, hence the latter part:

`ipv6_is_match(ip_part, "::1",0)`